libindy/src/services/crypto/mod.rs

```diff
@@ -296,6 +296,36 @@ impl CryptoService {
         Ok(decrypted_doc)
     }
 
+    pub fn authenticated_encrypt(&self, my_key: &Key, their_vk: &str, doc: &[u8]) -> Result<Vec<u8>, CryptoError> {
+        let msg = self.create_combo_box(my_key, their_vk, doc)?;
+
+        let msg_pack = msg.to_msg_pack()
+            .map_err(|e| CommonError::InvalidState(format!("Can't serialize ComboBox: {:?}", e)))?;
+
+        let res = self.encrypt_sealed(&their_vk, &msg_pack)?;
+
+        Ok(res)
+    }
+
+    pub fn authenticated_decrypt(&self, my_key: &Key, doc: &[u8]) -> Result<(String, Vec<u8>), CryptoError> {
```

Comment:
@kdenhartog

Hi,

Thanks for contribution. I suggest to perform some changes to make our terminology and internal API better:

You added new `authenticated_encrypt/authenticated_decrypt` method, but they don't perform what is usually mean on this in cryptography. Look to https://en.wikipedia.org/wiki/Authenticated_encryption. It defines output of authcrypt as a pair (ciphertext, mac). 

In crypto service we already have a function that provides similar output it is `encrypt`. It can be better to rename existing `encrypt/encrypt_sealed` to `authcrypt/anoncrypt`, but on other hand we already have conception on combo-box and i suggest to start using original NaCl names `box/box_open` and `seal_box/seal_box_open`

What `authenticated_encrypt` is actually doing is creation of combo box and i suggest to rename it to `combo_box` and `authenticated_decrypt` tp `combo_box_open`. Existing method `create_combo_box` doesn't look valuable and we don't have corresponded `open` pair.

As a result I suggest the following:

1. Rename `authenticated_encrypt/authenticated_encrypt` to `combo_box/combo_box_open`
2. Rename `encrypt/decrypt/encrypt_sealed/decrypt_sealed` to `box/box_open/sealed_box/sealed_box_open`
3. Remove existing `create_combo_box` method and just inline it.

@jovfer what do you think?

Reply:
@kdenhartog 

We discussed with Sergey that ```authenticated_encrypt``` is exposed as a public API. And it is better to keep it as is and remove number 1 from requested changes.

Reply:
All of these changes work except for the proposed "box" function name. This is a reserved keyword for rust it appears, so I've changed it to be "crypto_box" and "crypto_box_open". If we wanted to remain aligned with libsodium using this pattern, I'd suggest also changing "sealed_box" to "crypto_box_seal" and "sealed_box_open" to "crypto_box_seal_open". What are your thoughts on this additional change?